server/src/analyzer/typeAnalyzer.ts

```diff
@@ -175,6 +178,14 @@ export class TypeAnalyzer extends ParseTreeWalker {
             this.walk(node.suite);
         });
 
+        if (TypeUtils.isDataClass(classType)) {
+          classType.getClassFields().forEach((s, k) => {
+            if (!TypeUtils.isFunctionType(s.currentType) && /^_/.test(k) && s.declarations) {
+              this._addError(`Data field name can not start with _`, s.declarations[0].node);
```

Comment:
Is this just a convention, or is it enforced by the python interpreter? If it's just a convention, we may want to add a switch to enable/disable this. For now, it's fine to add a TODO comment.

Reply:
This is enforced in Python 3.7, results in runtime error.  Couldn't find official reference to this.  I've added a TODO regardless.